src/OrchardCore/OrchardCore.DisplayManagement/IShape.cs

```diff
@@ -3,16 +3,17 @@
 
 namespace OrchardCore.DisplayManagement
 {
-	/// <summary>
-	/// Interface present on dynamic shapes.
-	/// May be used to access attributes in a strongly typed fashion.
-	/// Note: Anything on this interface is a reserved word for the purpose of shape properties
-	/// </summary>
-	public interface IShape
-	{
-		ShapeMetadata Metadata { get; }
-		string Id { get; set; }
-		IList<string> Classes { get; }
-		IDictionary<string, string> Attributes { get; }
+    /// <summary>
+    /// Interface present on dynamic shapes.
+    /// May be used to access attributes in a strongly typed fashion.
+    /// Note: Anything on this interface is a reserved word for the purpose of shape properties
+    /// </summary>
+    public interface IShape
+    {
+        ShapeMetadata Metadata { get; }
+        string Id { get; set; }
+        IList<string> Classes { get; }
+        IDictionary<string, string> Attributes { get; }
+        IDictionary<object, object> Properties { get; }
```

Comment:
Why not `<string, object>`

Reply:
If it's because of Composite, then try to understand why it can't be a string indexer in composite first. And if not then still expose the string version for the interface compliance, even though it could be an object as the internal key.

Reply:
Because the underlying existing Properties Dictionary on `Composite` `DynamicObject` is `IDictionary<object, object>`.

Which is why I'm not totally sure we should use this, as Properties. Because it's exposing the Dictionary that has all the dynamic properties as well. 

Could be a `Data` Dictionary instead?

Reply:
Ah, crossing messages there. Sure I'll discover the underlying reason for `Composite`.

Reply:
I think we should remove the support for "indexes" then and just accept string parameters. Becuase the idea with Properties is exactly what the `_props` is, when you do `New.Foo(Prop1: 123)`

I don't think we ever need a property that is not a string. And supporting `New Foo()[123, 456]` where `123`  and `456` is a object index is not really useful. we could just hrow an excetpion if there are many values to `object [] indexes` or if the value is not a string.

Reply:
Done. Removed support for indexes, am not throwing an exception, just returning false from `TryGetIndex` which results in a `RuntimeBinderException`. Could also do it with a more detailed exception, but didn't want to throw in a `Try` method.

Did mean I was able to clean up some of the overrides from `ZoneHoldingBehaviour` as that was essentially managing the string as key already.